### PR TITLE
Add docx render option

### PR DESCRIPTION
**Pull request by avahoffman**
### Purpose/implementation Section

This PR is attempting to create a docx rendering option. This is mostly because I believe the GDSCN community would like to download student activities in word format so they can *have the option* be easily edited in whatever format suits them. I'm imagining two links at the bottom of the bookdown, "Download as .Rmd" and "Download as .docx"

#### What changes are being implemented in this Pull Request?

Addition of a docx rendering section in the rendering workflow

#### What was your approach?

Adding a rendering script. Not sure this is how we want it / is the best way to do this. Kind of feel it looks ugly compared to the other rendering, but this is just a first pass.

#### What GitHub issue does your pull request address?

N/A

### Tell potential reviewers what kind of feedback you are soliciting.

Does this fit with the quality of the template?

### New Content Checklist

- [ ] New content/chapter is in an Rmd file with [this kind of format and headers](https://github.com/jhudsl/DaSL_Course_Template_Bookdown/blob/main/02-chapter_of_course.Rmd).

- [ ] New content/chapter contains [Learning Objectives and are in the correct format](https://github.com/jhudsl/DaSL_Course_Template_Bookdown/wiki/Setting-up-images-and-graphics#learning-objectives-formatting).

- [ ] [Bookdown successfully re-renders and any new content files have been added to the _bookdown.yml](https://github.com/jhudsl/DaSL_Course_Template_Bookdown/wiki/Publishing-with-Bookdown).

- [ ] Spell check runs successfully in [Github actions style-n-check](https://github.com/jhudsl/DaSL_Course_Template_Bookdown/wiki/How-to-set-up-and-customize-GitHub-actions-robots#spell-check)).

- [ ] Any newly necessary packages that are needed have been added to the [Dockerfile and image](https://github.com/jhudsl/DaSL_Course_Template_Bookdown/wiki/Using-Docker#adding-packages-to-the-dockerfile).

- [ ] Images are in the [correct format for rendering](https://github.com/jhudsl/DaSL_Course_Template_Bookdown/wiki/Setting-up-images-and-graphics#adding-images-and-graphics-in-text).

- [ ] Every new image has [alt text and is in a Google Slide](https://github.com/jhudsl/DaSL_Course_Template_Bookdown/wiki/Setting-up-images-and-graphics#accessibility).

- [ ] Each slide is described in the notes of the slide so learners relying on a screen reader can access the content. See https://lastcallmedia.com/blog/accessible-comics for more guidance on this.

- [ ] The color palette choices of the slide are contrasted in a way that is friendly to those with color vision deficiencies.
You can check this using [Color Oracle](https://colororacle.org/).

### CODE REVIEW

**Comment by avahoffman** [comment]
This is some preliminary code to allow writing `.Rmd` to `.docx`. I think by default we'll have it off, but have it here in the template as an option.

I'm not experienced enough with github actions to know what's missing here. Basically, all checks pass and logs look fine but no docx files are created by github actions. Files are created fine locally. @cansavvy @carriewright11 would love your thoughts here! Hopefully I'm just missing something really obvious 🙂

**Comment by cansavvy** [comment]
@avahoffman thanks for doing this! This is a great start! I have a few things I think we need to add to wrap this up:

1) We need to add a comment to the preview.yml comment that brings someone to the rendered docx file.

2) I think it makes sense to make the script only run Rmds that are listed in the _bookdown.yml file. This ensures only Rmds that we want in the course are added to the docx. I could see scenarios where other Rmds are in the repository that aren't necessarily course material (example and template files and etc)

I'm happy to commit these changes to your branch if you want, or if you prefer to do them yourself, that's also fine. Just let me know.

**Comment by avahoffman** [comment]
@cansavvy agreed on all points! 

I did a 180 and made the docx creation part of the yml - it just renders the whole book as a docx based on _bookdown.yml - easier for everyone :) 

> We need to add a comment to the preview.yml comment that brings someone to the rendered docx file.

Not sure what you mean here - is there an equivalent of https://htmlpreview.github.io for other formats (docx?)

**Comment by cansavvy** [comment]
> Not sure what you mean here - is there an equivalent of https://htmlpreview.github.io for other formats (docx?)

Just something in that comment that brings people to the created file -- makes it clear that the file was made and where to find it. I can set this up if you like.

File: index.Rmd [hunk 1]
```diff
@@ -8,6 +8,9 @@ biblio-style: apalike
 link-citations: yes
 description: "Description about Course/Book."
 favicon: assets/dasl_favicon.ico
+output:
+    bookdown::word_document2:
+      toc: true
```
Comment (cansavvy) [inline index.Rmd:13]: Is this necessary in addition to the `output_format = 'all'`? Sometimes bookdown has things that are redundant and confusing. So just wondering.

**Comment by cansavvy** [comment]
Yay. Take a look at https://github.com/jhudsl/DaSL_Course_Template_Bookdown/pull/303#issuecomment-957802099 I definitely think this is something we should add to Release 1.2 #300 I think we can almost merge it (if you are ready). 

We might want to think about how we advise this docx is used though. We could probably be fancy and have changes to the docx uploaded to Google Drive, but I'd also hate for people's edits to get overwritten at any point, so I'm not sure what's the ideal workflow for this. I'll make an issue for us to discuss this. 

Edit: see #306

File: index.Rmd (hunk 1 above)
Comment (avahoffman) [reply index.Rmd:13]: Locally, `bookdown::render_book(output_format = "all")` renders the docx but `bookdown::render_book()` doesn't. I agree it's weird and kind of redundant.

Comment (avahoffman) [reply index.Rmd:13]: From the documentation:

> If using NULL then the output format is the first one defined in the YAML frontmatter in the input file (this defaults to HTML if no format is specified there).

**Comment by avahoffman** [comment]
> Just something in that comment that brings people to the created file -- makes it clear that the file was made and where to find it. I can set this up if you like.

Thanks for adding! I didn't think to reference the preview branch 🙂

**Comment by carriewright11** [comment]
Good job to both of you! I will check out the discussion issue that @cansavvy created.

File: index.Rmd (hunk 1 above)
Comment (cansavvy) [reply index.Rmd:13]: I'm always so confused by the various places where yamls are specified and how to know which is prioritized. But if you have it figured out then sounds good to me! :+1:

**Review by cansavvy — APPROVED** [review]
Can I mark this PR as ready and merge it then, @avahoffman ?

**Comment by cansavvy** [comment]
The Render preview fail is just a temporary problem due to Docker's server. Nothing crazy. Should be resolved if we re-run in a tad.

File: index.Rmd (hunk 1 above)
Comment (avahoffman) [reply index.Rmd:13]: haha me too - it took me forever to find that bit in the docs :P

**Comment by avahoffman** [comment]
Woo! Thanks for working with me here! It's the remote dev I love to see 😄

**Pull request merged**